src/platform/packages/shared/shared-ux/code_editor/impl/code_editor.tsx

```diff
@@ -851,18 +861,25 @@ const useFitToContent = ({
  * @description See {@link https://github.com/elastic/kibana/issues/223981} for the rationale behind this bug fix implementation
  */
 const UseBug223981FixRepositionSuggestWidget: FC<
-  PropsWithChildren<{ editor: monaco.editor.IStandaloneCodeEditor | null }>
-> = ({ children, editor }) => {
+  PropsWithChildren<{
+    editor: monaco.editor.IStandaloneCodeEditor | null;
+    enableSuggestWidgetRepositioning: boolean;
+  }>
+> = ({ children, editor, enableSuggestWidgetRepositioning }) => {
   const { euiTheme } = useEuiTheme();
   const suggestWidgetModifierClassName = 'kibanaCodeEditor__suggestWidgetModifier';
 
   useEffect(() => {
     // @ts-expect-errors -- "widget" is not part of the TS interface but does exist
     const suggestionWidget = editor?.getContribution('editor.contrib.suggestController')?.widget
       ?.value;
-
     // The "onDidShow" and "onDidHide" is not documented so we guard from possible changes in the underlying lib
-    if (suggestionWidget && suggestionWidget.onDidShow && suggestionWidget.onDidHide) {
+    if (
+      suggestionWidget &&
+      suggestionWidget.onDidShow &&
+      suggestionWidget.onDidHide &&
+      enableSuggestWidgetRepositioning
+    ) {
```

Comment:
Changes look good to me, I've got just a nit since we are conditionally making calculations here, we could also only insert the style declaration on L912 - L919 conditionally